src/plugin.ts

```diff
@@ -83,24 +104,90 @@ export default function mediawikiUserscript( options: PluginOptions ): Plugin {
 						config.build.rollupOptions.output.manualChunks = undefined;
 					}
 				} else {
-					config.build.rollupOptions.output = { manualChunks: undefined };
+					config.build.rollupOptions.output = {
+						manualChunks: undefined
+					};
 				}
 
 				if ( !optimizeDepsEntries ) {
 					// Indicate this as the entrypoint of our script
 					config.optimizeDeps.entries = [ options.entry ];
 				}
-				config.optimizeDeps.exclude =
-					Array.isArray( config.optimizeDeps.exclude ) ?
-						[ ...config.optimizeDeps.exclude, ...( externals ) ] :
-						externals;
+				config.optimizeDeps.exclude = Array.isArray( config.optimizeDeps.exclude ) ?
+					[ ...config.optimizeDeps.exclude, ...externals ] :
+					externals;
 			}
 
 			getHook( cssPlugin.config ).call( this, config, env );
 		},
 		configResolved( config ) {
 			getHook( cssPlugin.configResolved ).call( this, config );
 		},
+		resolveId( source, importer ) {
+			if ( externals.includes( source ) ) {
+				importers.add( importer );
+				return EXTERN_PREFIX + source;
+			}
+		},
+		load( id ) {
+			if ( id.startsWith( EXTERN_PREFIX ) ) {
+				const name = id.slice( EXTERN_PREFIX.length );
+
+				let code = '';
+				if ( options.resourceLoaderDebugCookieAge ) {
+					code += `document.cookie = "resourceLoaderDebug=2;max-age=${options.resourceLoaderDebugCookieAge}"\n`;
+				}
+				code += `const __mw_module = (await mw.loader.using(${JSON.stringify( name )}))(${JSON.stringify( name )});\n`;
+				code += 'export default __mw_module;\n';
+				if ( name === 'vue' ) {
+					code +=
+						'if (typeof __VUE_HMR_RUNTIME__ !== "object") {\n' +
+						'console.error("[vite-mw-userscript] ResourceLoader debug mode is not enabled. Without it, ' +
```

Comment:
This is almost always going to be wrong on first run if `resourceLoaderDebugCookieAge` is set, since the cookie wouldn't have been set at the time of page load (where Vue gets loaded, on some pages). Maybe we should use a different message for when it's set but `__VUE_HMR_RUNTIME__` is missing?

Reply:
Fixed in 6812886